explorer/interpreter/interpreter.cpp

```diff
@@ -1697,10 +1698,9 @@ auto Interpreter::StepStmt() -> ErrorOr<Success> {
             &cast<For>(stmt).variable_declaration()));
       }
       if (act.pos() == 2) {
-        Nonnull<const BindingPlaceholderValue*> loop_var =
-            cast<const BindingPlaceholderValue>(
-                act.results()[LoopVarPosInResult]);
-        Nonnull<const TupleValue*> source_array =
+        auto loop_var = cast<const BindingPlaceholderValue>(
+            act.results()[LoopVarPosInResult]);
+        auto source_array =
             cast<const TupleValue>(act.results()[TargetVarPosInResult]);
```

Comment:
```suggestion
        const auto* loop_var = cast<BindingPlaceholderValue>(
            act.results()[LoopVarPosInResult]);
        const auto* source_array =
            cast<TupleValue>(act.results()[TargetVarPosInResult]);
```
Can we ask clang-tidy to do this? Particularly, using `auto*` rather than `auto` when the variable is a pointer. I think it has a configuration setting for that.

Reply:
https://clang.llvm.org/extra/clang-tidy/checks/readability/qualified-auto.html

Reply:
Please feel free to set new flags, but I think it's outside the scope here

Reply:
I can back out fixes if preferred, I just don't want to get into a long tail of work